lib/code_corps/github/sync/sync.ex

```diff
@@ -99,18 +100,54 @@ defmodule CodeCorps.GitHub.Sync do
     |> transact()
   end
 
-  def sync_issues(repo) do
+  def sync_repo(repo) do
+    {:ok, pulls} = GitHub.API.Repository.pulls(repo)
+    github_pull_requests = Enum.map(pulls, &sync_pull(&1, repo))
+    results = sync_issues(repo, github_pull_requests)
+    github_issues = for {:ok, %{github_issue: github_issue}} <- results, do: github_issue
+    github_issues = Enum.map(github_issues, fn issue -> {:ok, issue} end)
+    tasks = for {:ok, %{tasks: tasks}} <- results, do: tasks
+    tasks = tasks |> List.flatten
+    tasks = for %CodeCorps.Task{} = task <- tasks, do: task
+    tasks = Enum.map(tasks, fn task -> {:ok, task} end)
+    sync_comments(repo, github_issues, tasks)
+    repo
+  end
```

Comment:
Inside here we could sync each step, modify the GithubRepo sync state, and then return the repo when done.